src/test/java/org/cactoos/io/LoggingInputStreamTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.io;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.IsEqual;
+import org.hamcrest.core.IsNull;
+import org.junit.Test;
+import org.llorllale.cactoos.matchers.Assertion;
+
+/**
+ * Test case for {@link LoggingInputStream}.
+ *
+ * @since 0.39
+ * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle MagicNumberCheck (500 lines)
+ */
+public final class LoggingInputStreamTest {
+
+    @Test(expected = Exception.class)
+    public void reThrowsException() throws IOException {
+        final LoggingInputStream stream = new LoggingInputStream(
+            new InputStream() {
+                @Override
+                public int read() throws IOException {
+                    throw new IOException("Some read exception");
+                }
+            },
+            this.getClass().getSimpleName()
+        );
+        MatcherAssert.assertThat(
```

Comment:
@ilyakharlamov Please, use `Assertion` instead of `MatcherAssert.assertThat`.

Reply:
`new Assertion(..() -> throw new Exception() ).affirm`, that chews up the actual exception so it will no longer be possible to check if the exception is re-thrown at `@Test(expected....)`

Reply:
@fabriciofx This is a [2nd time](https://github.com/yegor256/cactoos/pull/1028#discussion_r249615042). How do you want me to test for an exception using that freaking`Throws` matcher? The expected usage (from cactoos-matchers unit tests) is `new Throws<>("description", Exception.class).matchesSafely(...)` but the `matchesSafely(..)` is private, so what do you want me to do?

Reply:
@ilyakharlamov I was thinking something like:
```
final LoggingInputStream stream = new LoggingInputStream(
    new InputStream() {
        @Override
        public int read() throws IOException {
            throw new IOException("Some read exception.");
        }
    },
    this.getClass().getSimpleName()
);
new Assertion<>(
    "Read doesn't throw an the exception.",
    () -> () -> stream.read(),
    new Throws<>("Some read exception.", IOException.class)
).affirm();
```
Is it possible?

Reply:
> @fabriciofx This is a [2nd time](https://github.com/yegor256/cactoos/pull/1028#discussion_r249615042). How do you want me to test for an exception using that freaking`Throws` matcher? The expected usage (from cactoos-matchers unit tests) is `new Throws<>("description", Exception.class).matchesSafely(...)` but the `matchesSafely(..)` is private, so what do you want me to do?

@ilyakharlamov I thought this comment sounded a bit rude. We don't need this, right?

Reply:
@fabriciofx I am sorry if that sounded rude, I didn't want to offend anyone. Just the policy looks too strict and vague at the same time. How would a stranger knew that Throws needs to be used here, all other tests use `@Test(expected)`?

Reply:
@ilyakharlamov Right, no problem. But this is the reviewer's role: looking for things like that and show the correct way. And I understand you. The question is `@Test(expected)` is too procedural and will be removed (there's an issue about that) but to new code, the orientation is write using `cactoos-matchers` API (`Assertion`, `Throws` and others).